### DISCUSSION TITLE
API changes in master branch

### DISCUSSION THREAD

**Discussion by scpeters** [bulletphysics/bullet3#3487] [General]
As an FYI to other users, the recently merged #363 made some API changes.

In f4b0cc85cdf5a6f8292bbd1b50b57d3fb3b02d0d for example, the `btHinge2Constraint` class is changed to derive from `btGeneric6DofSpring2Constraint` instead of `btGeneric6DofSpringConstraint`. The main issue is that now the [getRotationalLimitMotor function returns type btRotationalLimitMotor2](https://github.com/bulletphysics/bullet3/blob/master/src/BulletDynamics/ConstraintSolver/btGeneric6DofSpring2Constraint.h#L331) which [does not inherit from btRotationalLimitMotor](https://github.com/bulletphysics/bullet3/blob/master/src/BulletDynamics/ConstraintSolver/btGeneric6DofSpring2Constraint.h#L69). We can work around this with `ifdef`'s if backward compatibility can't be achieved.

**Comment by erwincoumans** [comment]
The btGeneric6DofSpringConstraint is going to be deprecated and removed in the future. Is there a particular feature/API thing missing from the new btGeneric6DofSpring2Constraint and/or btRotationalLimitMotor2? Thanks for the feedback!

**Comment by scpeters** [comment]
I haven't checked for missing features. I was just noticing a failed gazebo build, so we'll probably have to use ifdef's until bullet 2.83 is available on Ubuntu platforms.

**Comment by bchretien** [comment]
:+1: Gazebo compilation is currently broken on Arch Linux (Bullet 2.83) because of this.
